### ISSUE TITLE
OnDidChange TextEvent values not set correctly upon single newline character.

### ISSUE THREAD

**Issue by james-y** [atom/atom#9842]
### Reproduce:

Subscribe to the TextEditor OnDidChange event and simply output the TextEvent object to the console log.
Enter a newline into a new text buffer in the UI (Hit Enter/Return) to fire off the OnDidChange event.
Examine the logged console object:
1. The TextEvent object field 'newText' does not include the newline character, it is an empty String.
2. The oldRange values are incorrect for the end/start fields.  They show the same position as the newRange.

Compare this to a backspace/delete that removes the newline and moves the cursor up one row.
The TextEvent object field 'oldText' includes the newline character that was deleted.
I think the two operations should be inverses, and the values of oldText/newText fields mirrors of each other in either situation at the very least.

_Update:_
The TextEvent does fire, but an additional TextEvent with invalid information also occurs.

**Atom Version:** 1.2.4
**System:** Ubuntu 14.04.3
### Done When:

The additional event that describes no change (newText and oldText both empty Strings) does not occur.

**Comment by james-y** [comment]
It looks like two events occur on a newline character being added.  One has the correct information in the TextEvent object, the other is incorrect.

**Comment by cynicaldevil** [comment]
Does this still need work? Should I take it up?

**Comment by lee-dohm** [comment]
It's still open, so if you can still reproduce it, then yes feel free :grinning:

**Comment by cynicaldevil** [comment]
Well I am able to replicate this bug, and whenever I type in a newline, the `onDidChange` method is called two times. But the object being passed to the callback function doesn't have keys like `oldText` or `newText` like I expected. Instead it has keys like `start`, `end`, `screenDelta` and `bufferDelta`. Why is this so?

**Comment by cynicaldevil** [comment]
Oh well, ignore my previous comment, I sorted that out.
@james-y 
I found that there are two cases which can occur when entering a newline char:
**1. when you enter a newline on a line which has text in it**
I think you were referring to this case in your original issue. Here, the text editor calls the `onDidChange` method twice: one for the new line char, and another one for the spaces/tabs it *automatically* inserts to auto-indent the new line properly. If there's no auto indent, the second textEvent object simply contains an empty string in its `newText` field. I think this is logical enough.

**2. when a newline is entered on a blank line**
In this case, the `onDidChange` method is called three times: one with the correct info, and other two containing empty strings in their `newText` field. I think I will work on this.
